pkg/admission/pathannotation/pathannotation_admission_test.go

```diff
@@ -0,0 +1,313 @@
+				if err != nil {
+					t.Fatal(err)
+				}
+				if _, has := objMeta.GetAnnotations()[core.LogicalClusterPathAnnotationKey]; has {
+					t.Fatalf("the %q annotation cannot be automatically set on an APIResourceSchema resource", core.LogicalClusterPathAnnotationKey)
+				}
+			},
+		},
+		{
+			name:                    "admission is applied to an unsupported resource if it has the path annotation present",
+			admissionContext:        wellKnownAdmissionContext,
+			admissionResource:       apisv1alpha1.SchemeGroupVersion.WithResource("apiresourceschemas"),
+			admissionVerb:           admission.Create,
+			admissionObject:         &apisv1alpha1.APIResourceSchema{ObjectMeta: metav1.ObjectMeta{Annotations: map[string]string{core.LogicalClusterPathAnnotationKey: ""}}},
+			getLogicalCluster:       wellKnownGetLogicalCluster,
+			validateAdmissionObject: wellKnownAdmissionObjectValidator,
+		},
+		{
+			name:              "a path is derived from the LogicalCluster object if it doesn't have the path annotation",
+			admissionContext:  wellKnownAdmissionContext,
+			admissionVerb:     admission.Create,
+			admissionResource: apisv1alpha1.SchemeGroupVersion.WithResource("apiexports"),
+			admissionObject:   &apisv1alpha1.APIExport{},
+			getLogicalCluster: func(clusterName logicalcluster.Name, name string) (*corev1alpha1.LogicalCluster, error) {
+				lc, err := wellKnownGetLogicalCluster(clusterName, name)
+				if err != nil {
+					return nil, err
+				}
+				lc.Annotations[core.LogicalClusterPathAnnotationKey] = "universe:milky-way"
+				return lc, nil
+			},
+			validateAdmissionObject: func(t *testing.T, obj runtime.Object) {
+				objMeta, err := meta.Accessor(obj)
+				if err != nil {
+					t.Fatal(err)
+				}
+				pathAnnotation := objMeta.GetAnnotations()[core.LogicalClusterPathAnnotationKey]
+				if pathAnnotation != "universe:milky-way" {
+					t.Fatalf("unexpected value = %q, in the %q annotation, expected = %q", pathAnnotation, core.LogicalClusterPathAnnotationKey, "universe:milky-way")
+				}
+			},
+		},
+		{
+			name:                    "a path is updated when is different from the one applied to the LogicalCluster resource",
+			admissionVerb:           admission.Create,
+			admissionResource:       apisv1alpha1.SchemeGroupVersion.WithResource("apiexports"),
+			admissionObject:         &apisv1alpha1.APIExport{ObjectMeta: metav1.ObjectMeta{Annotations: map[string]string{core.LogicalClusterPathAnnotationKey: "bar:foo"}}},
+			admissionContext:        wellKnownAdmissionContext,
+			getLogicalCluster:       wellKnownGetLogicalCluster,
+			validateAdmissionObject: wellKnownAdmissionObjectValidator,
+		},
+		{
+			name:                    "happy path: an APIExport is annotated with a path",
+			admissionVerb:           admission.Create,
+			admissionResource:       apisv1alpha1.SchemeGroupVersion.WithResource("apiexports"),
+			admissionObject:         &apisv1alpha1.APIExport{},
+			admissionContext:        wellKnownAdmissionContext,
```

Comment:
nit: factoring out the data that drives the test would make this easier to read (and explicit, not implicit):

```go
		{
			name:                    "happy path: an APIExport is annotated with a path",
			admissionVerb:           admission.Create,
			admissionResource:       apisv1alpha1.SchemeGroupVersion.WithResource("apiexports"),
			admissionObject:         &apisv1alpha1.APIExport{},
			admissionContext:        admissionContextFor("root:foo"),
			getLogicalCluster:       getCluster("root:foo"),
			validateAdmissionObject: objectHasPathAnnotation("root:foo"),
		},
```

Reply:
done